pkg/devfile/parse_test.go

```diff
@@ -751,12 +775,38 @@ spec:
 				}
 				return nil
 			},
+			wantError: false,
+		},
+		{
+			name: "with unsupported schema version",
+			args: args{
+				args: parser.ParserArgs{
+					ExternalVariables: map[string]string{
+						"PARAMS": "baz",
+					},
+					Data: []byte(devfileContentWithUnsupportedSchema),
+				},
+			},
+			wantCommandLine: "./main baz",
+			wantVariables: map[string]string{
+				"PARAMS": "baz",
+			},
+			wantVarWarning: variables.VariableWarning{
+				Commands:        map[string][]string{},
+				Components:      map[string][]string{},
+				Projects:        map[string][]string{},
+				StarterProjects: map[string][]string{},
+			},
+			wantError: true,
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			gotD, gotVarWarning, err := ParseDevfileAndValidate(tt.args.args)
-			if err != nil {
+			if err != nil && tt.wantError {
+				t.Log("Correctly identified unsupported schema version")
```

Comment:
In the event we want to support other expected errors, could we add something like `wantErrStr` to see which error should be expected?

Reply:
Yeah I agree `wantErrStr` would be more appropriate here so we can verify the correct errors are occurring as well. Let me make some changes to it